flocker/node/_docker.py

```diff
@@ -543,14 +543,26 @@ def _list():
                     # have been removed in another thread.
                     if e.response.status_code == NOT_FOUND:
                         continue
+                    else:
+                        raise
 
                 state = (u"active" if data[u"State"][u"Running"]
                          else u"inactive")
                 name = data[u"Name"]
                 image = data[u"Config"][u"Image"]
                 command = data[u"Config"][u"Cmd"]
-                image_command = self._client.inspect_image(image)
-                if image_command[u"Config"][u"Cmd"] == command:
+                try:
+                    image_data = self._client.inspect_image(image)
+                except APIError as e:
+                    if e.response.status_code == NOT_FOUND:
+                        # Image has been deleted, so just fill in some
+                        # stub data so we can return *something*. This
+                        # should happen only for stopped containers so
+                        # some inaccuracy is acceptable.
+                        image_data = {u"Config": {u"Env": [], u"Cmd": []}}
```

Comment:
I believe we can get the underling Docker image hash from the inspect results, so we could use that instead, solving the tagging issue. I will try to do that.